components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
+        const reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit
+        );
+        if (get(reddit_things, "data.after") !== undefined) {
+          after = reddit_things.data.after;
+          console.log(
+            `after children length: ${reddit_things.data.children.length}`
+          );
+          if (reddit_things.data.children.length > 0) {
+            reddit_things.data.children.forEach((reddit_link) => {
+              this.emitMeInfo(reddit_link);
+            });
+          }
+        }
+      } while (after);
+      this.db.set("after", after);
+      console.log(`finished [deploy] hook, after: ${after}`);
+    },
+  },
+  methods: {
+    emitMeInfo(reddit_link) {
+      this.$emit(reddit_link, {
+        summary: reddit_link.data.name,
+      });
+    },
+  },
+  async run() {
+    // Emmits new events as per timer config
+
+    let current_after = this.db.get("after");
+    console.log(`started [run] method, after: ${current_after}`);
+    const reddit_things = await this.reddit.getNewSubredditLinks(
+      current_after,
+      this.subreddit
+    );
+
+    if (get(reddit_things, "data.after") !== undefined) {
+      var new_after = reddit_things.data.after;
+      if (new_after) {
+        if (reddit_things.data.children.length > 0) {
+          reddit_things.data.children.forEach((reddit_link) => {
+            this.emitMeInfo(reddit_link);
+          });
+          this.db.set("after", new_after);
+          current_after = new_after;
+        }
+      }
+    }
+    console.log(`finished [run] method, after: ${current_after}`);
```

Comment:
When we call `this.db.set`, we'll automatically write a log that shows the value written to the DB, so you don't need this `console.log` statement or this line above:

```javascript
current_after = new_after;
```

Reply:
Yes, I noticed that.
Just thought it was easier to debug with a more human-friendly considering I had sinking the component model knowledge. But yeah, I will keep it to let them use auto log when merging by removing in case I had used it.